milli/src/search/new/matches/mod.rs

```diff
@@ -800,6 +797,32 @@ mod tests {
         );
     }
 
+    #[test]
+    fn format_highlight_crop_phrase_query() {
+        //! testing: https://github.com/meilisearch/meilisearch/issues/3975
+        let temp_index = temp_index_with_documents();
+        let rtxn = temp_index.read_txn().unwrap();
+
+        let format_options = FormatOptions { highlight: true, crop: Some(10) };
+        let text = "The groundbreaking invention had the power to split the world between those who embraced progress and those who resisted change!";
+
+        let builder = MatcherBuilder::new_test(&rtxn, &temp_index, "\"the world\"");
+        let mut matcher = builder.build(text);
+        // should return 10 words with a marker at the start as well the end, and the highlighted matches.
+        insta::assert_snapshot!(
+            matcher.format(format_options),
+            @"…had the power to split <em>the</em> <em>world</em> between those who…"
+        );
+
+        let builder = MatcherBuilder::new_test(&rtxn, &temp_index, "those \"and those\"");
+        let mut matcher = builder.build(text);
+        // should highlight both "and" and "those".
+        insta::assert_snapshot!(
+            matcher.format(format_options),
+            @"…between those who embraced progress <em>and</em> <em>those</em> who resisted change…"
```

Comment:
In my sense the first `those` should be highlighted, because it matches the first `those` 🤔 
```suggestion
            @"…between <em>those</em> who embraced progress <em>and</em> <em>those</em> who resisted change…"
```

Could you try to fix it, please? 😄

Reply:
The logic looks correct. I created a new index and inserted this document:
```json
{
	"id": 1,
	"title": "test-pr",
	"overview": "The groundbreaking invention had the power to split the world between those who embraced progress and those who resisted change!"
}
```
Then, I ran a search for phrase `"those \"and those\""`:
```json
{
	"q": "those \"and those\"",
	"attributesToHighlight": [
		"overview"
	],
	"attributesToCrop": [
		"overview"
	],
	"cropLength": 10
}
```

I get the following response back, which highlights the first `those` as well the phrase `and those`:
```json
{
	"hits": [
		{
			"id": 1,
			"title": "test-pr",
			"overview": "The groundbreaking invention had the power to split the world between those who embraced progress and those who resisted change!",
			"_formatted": {
				"id": "1",
				"title": "test-pr",
				"overview": "…world between <em>those</em> who embraced progress <em>and</em> <em>those</em> who resisted…"
			}
		}
	],
	"query": "those \"and those\"",
	"processingTimeMs": 4,
	"limit": 20,
	"offset": 0,
	"estimatedTotalHits": 1
}
```

Reply:
@ManyTheFish I created a new temporary index in the test and added the below document to it:
```json
{ "id": 1, "text": "The groundbreaking invention had the power to split the world between those who embraced progress and those who resisted change!" }
```
Now the test correctly highlights the first `those` as well the phrase `and those`.

Let me know if this correct or not 😅